.idea/misc.xml

```diff
@@ -0,0 +1,62 @@
+      <item index="5" class="java.lang.String" itemvalue="io.druid.annotations.UsedInGeneratedCode" />
+      <item index="6" class="java.lang.String" itemvalue="io.druid.guice.annotations.ExtensionPoint" />
+      <item index="7" class="java.lang.String" itemvalue="io.druid.guice.annotations.PublicApi" />
+      <item index="8" class="java.lang.String" itemvalue="io.druid.java.util.common.lifecycle.LifecycleStart" />
+      <item index="9" class="java.lang.String" itemvalue="io.druid.java.util.common.lifecycle.LifecycleStop" />
+      <item index="10" class="java.lang.String" itemvalue="javax.inject.Inject" />
+      <item index="11" class="java.lang.String" itemvalue="org.openjdk.jmh.annotations.Benchmark" />
+    </list>
+    <writeAnnotations>
+      <writeAnnotation name="com.fasterxml.jackson.annotation.JacksonInject" />
+      <writeAnnotation name="com.fasterxml.jackson.annotation.JsonProperty" />
+      <writeAnnotation name="com.google.caliper.Param" />
+      <writeAnnotation name="io.airlift.airline.Option" />
+      <writeAnnotation name="org.openjdk.jmh.annotations.Param" />
+    </writeAnnotations>
+  </component>
+  <component name="MavenProjectsManager">
+    <option name="originalFiles">
+      <list>
+        <option value="$PROJECT_DIR$/pom.xml" />
+      </list>
+    </option>
+  </component>
+  <component name="NullableNotNullManager">
+    <option name="myDefaultNullable" value="javax.annotation.Nullable" />
+    <option name="myDefaultNotNull" value="org.jetbrains.annotations.NotNull" />
+    <option name="myNullables">
+      <value>
+        <list size="6">
+          <item index="0" class="java.lang.String" itemvalue="org.jetbrains.annotations.Nullable" />
+          <item index="1" class="java.lang.String" itemvalue="javax.annotation.Nullable" />
+          <item index="2" class="java.lang.String" itemvalue="javax.annotation.CheckForNull" />
+          <item index="3" class="java.lang.String" itemvalue="org.springframework.lang.Nullable" />
+          <item index="4" class="java.lang.String" itemvalue="edu.umd.cs.findbugs.annotations.Nullable" />
+          <item index="5" class="java.lang.String" itemvalue="android.support.annotation.Nullable" />
+        </list>
+      </value>
+    </option>
+    <option name="myNotNulls">
+      <value>
+        <list size="4">
+          <item index="0" class="java.lang.String" itemvalue="org.jetbrains.annotations.NotNull" />
+          <item index="1" class="java.lang.String" itemvalue="javax.annotation.Nonnull" />
+          <item index="2" class="java.lang.String" itemvalue="edu.umd.cs.findbugs.annotations.NonNull" />
+          <item index="3" class="java.lang.String" itemvalue="android.support.annotation.NonNull" />
+        </list>
+      </value>
+    </option>
+  </component>
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="false" project-jdk-name="1.8" project-jdk-type="JavaSDK">
```

Comment:
To avoid conflicts, developers will need to rename their JDK to "1.8", it could be done in IntelliJ in "Project Structure..." -> "Platform Settings" -> "SDKs".

Reply:
Hmm, I don't think fixing the project-jdk-name is a good idea. It's possible that some developers have installed two or more JDKs in their machines. Suggest to remove this.

Reply:
> It's possible that some developers have installed two or more JDKs in their machines.

I don't see why multiple JDKs is the problem.

> Suggest to remove this.

The problem is that IntelliJ will add this line automatically in the file whenever it can, so you will always have "unchecked" changes in misc.xml, from Git point of view.

Reply:
> I don't see why multiple JDKs is the problem.

JDK configurations are shared across multiple projects, and multiple JDKs should have different names which are usually intuitive like ```1.8 oracle```. If there is no JDK of name ```1.8```, Intellij will fail to find a JDK automatically. Finally, developers should set one of their JDKs as the project SDK which modifies this file, or rename one of their JDKs to ```1.8```. I think both cases are not good.

Reply:
Developers could also add "another" JDK which is called `1.8` and points to the same files as some existing JDK, even if it's 1.9 in fact. `1.8` is just an alias name. I agree that it's not ideal, but I don't see a better decision

Reply:
Added a section to INTELLIJ_SETUP.md

Reply:
Thanks.